lib/utils/extension/get_it_extension.dart

```diff
@@ -0,0 +1,9 @@
+import 'package:get_it/get_it.dart';
+
+C? getIt<C extends Object>({C Function()? builder}) {
+  if (!GetIt.I.isRegistered<C>()) {
+    if (builder == null) return null;
+    GetIt.I.registerLazySingleton<C>(builder);
+  }
+  return GetIt.I.get<C>();
+}
```

Comment:
Because this way we don't need to do `final object =getIt.get<Object>();` everytime we need a registered class.
Just use the instance of the class and it will registered if it did not already and then return the instance.

Also the allow to register a class inside GetIt only when we need it

Reply:
For example: 
https://github.com/linagora/twake-on-matrix/blob/abe5f18f87eb7d7064fc9a77ab38b984d96f9cc2/lib/presentation/notifiers/active_room_id_notifier.dart

https://github.com/linagora/twake-on-matrix/blob/abe5f18f87eb7d7064fc9a77ab38b984d96f9cc2/lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body.dart